.github/workflows/latest_deps.yml

```diff
@@ -0,0 +1,160 @@
+# People who are freshly `pip install`ing from the wheels we publish to PyPI will pull
+# in the latest versions of all dependencies. We used to do this in our CI runs, which
+# would detect any incompatibilities due to new package releases. Now that we use the
+# locked poetry environment, we don't have advance notice of these breakages.defaults:
+#
+# To plug that hole, this job is a canary which periodically
+# - checks out develop,
+# - pip installs from source (building a wheel locally using `poetry-core`,
+#   and `pip install`ing from that), and
+# - runs test suites against that checkout.
+#
+# Based on the twisted trunk CI job.
+
+name: Latest dependencies
+
+on:
+  schedule:
+    - cron: 0 7 * * *
+  pull_request:
+    branches: [dmr/poetry-wheel-ci]
+  push:
+    branches: [dmr/poetry-wheel-ci]
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  # TODO(dmr): I'm not sure if we want to run mypy here, nor if mypy failures should
+  # merit an issue.
```

Comment:
I think it is fine, we've seen breakages from people adding type hints to our dependencies.

Reply:
Thanks. I'll remove the TODO.

```suggestion
```